Dockerfile

```diff
@@ -1,23 +1,32 @@
 # Base image
-FROM nvidia/cuda:11.8.0-cudnn8-runtime-ubuntu20.04
+FROM nvidia/cuda:11.8.0-cudnn8-runtime-ubuntu22.04
+
+ENV DEBIAN_FRONTEND noninteractive
+
+# Install system dependencies
+RUN apt-get update && \
+    apt-get install -y python3-pip libpq-dev libgdal-dev gdal-bin
+
+# Verify GDAL version
+RUN if [ $(gdal-config --version | cut -d '.' -f1,2) \< "3.1" ]; then echo $(gdal-config --version) && exit 1; fi
+
+# Install Poetry
+RUN pip3 install poetry
```

Comment:
Do not invoke `pip3` like this - always invoke as `python3.x -m pip` to ensure you are using the correct python!

Reply:
Alright will do. This image uses Python 3.10